src/Components/VideoCard/VideoCard.jsx

```diff
@@ -0,0 +1,58 @@
+import {useState} from "react";
+import { Button } from "../index";
+import { BsThreeDotsVertical, MdOutlineWatchLater, MdPlaylistAdd } from "../../assets";
+
+export const VideoCard = (props) => {
+    const [showCtaBox, setShowCtaBox] = useState(false);
+
+    const {
+        _id,
+    title,
+    creator,
+    categoryName,
+    views,
+    postedBefore
+    } = props.videoDetails;
+
+    const calculateView = (views) => {
+        // Nine Zeros for Billions
+        return Math.abs(Number(views)) >= 1.0e+9 
+        ? (Math.abs(Number(views)) / 1.0e+9).toFixed(1) + "B"
+        // Six Zeroes for Millions 
+        : Math.abs(Number(views)) >= 1.0e+6
+        ?( Math.abs(Number(views)) / 1.0e+6).toFixed(1) + "M"
+        // Three Zeroes for Thousands
+        : (Math.abs(Number(views))) >= 1.0e+3
+        ? (Math.abs(Number(views)) / 1.0e+3).toFixed(1) + "K"
+        : Math.abs(Number(views));
```

Comment:
Appreciate the thought here, but why haven't you saved `Math.abs(Number(views))` in a variable and then use it